src/lib/tvgPictureImpl.h

```diff
@@ -188,18 +199,32 @@ struct Picture::Impl
         if (loader) loader->close();
         loader = LoaderMgr::loader(data, w, h, copy);
         if (!loader) return Result::NonSupport;
+        w = loader->w;
+        h = loader->h;
+        vw = loader->vw;
+        vh = loader->vh;
         return Result::Success;
     }
 
     Paint* duplicate()
     {
         reload();
 
-        if (!paint) return nullptr;
+        if (!paint && !pixels) return nullptr;
+
         auto ret = Picture::gen();
         if (!ret) return nullptr;
         auto dup = ret.get()->pImpl;
-        dup->paint = paint->duplicate();
+        dup->w = w;
+        dup->h = h;
+        dup->vw = vw;
+        dup->vh = vh;
+        dup->vx = vx;
+        dup->vy = vy;
+        dup->preserveAspect = preserveAspect;
+
+        if (paint) dup->paint = paint->duplicate();
+        else if (pixels) dup->pixels = pixels;
```

Comment:
is it ok? 

is it possible, that the main Picture will be deleted before the duplicated one? then I'll copy the content